app/react/ToggledFeatures/tocGeneration/specs/ReviewTocButton.spec.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { shallow, ShallowWrapper } from 'enzyme';
+import configureStore, { MockStore, MockStoreCreator } from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import { ClientFile } from 'app/istore';
+import { ReviewTocButton } from '../ReviewTocButton';
+
+describe('ReviewTocButton', () => {
+  let component: ShallowWrapper<typeof ReviewTocButton>;
+
+  const mockStoreCreator: MockStoreCreator<object> = configureStore<object>([]);
+  const render = (file: Partial<ClientFile>) => {
+    const store: MockStore<object> = mockStoreCreator({});
+    component = shallow(
+      <Provider store={store}>
+        <ReviewTocButton file={{ ...file, _id: 'id' }}>
+          <span>test</span>
+        </ReviewTocButton>
+      </Provider>
+    )
+      .dive()
+      .dive();
```

Comment:
We have been requiring this wrapper with store and dual `dive()` calls on a lot of new components.  Just like an `idea`, we may want to look into abstracting this.

Reply:
we can think about it, the problem i faced when i have tried this is that the number of dives can vary and sometime you want the context or not in the dive, as an example:

https://github.com/huridocs/uwazi/blob/37d91765babc2e6b685a584d39f10e578777c2e7/app/react/Templates/components/specs/MetadataProperty.spec.js#L80-L92

but maybe this is wrong and the 2 dive scenario with Provider->Component is generic enough ? should i create a tech debt issue for this ?

Reply:
Let's wait and see if they keep popping up. Let's keep an eye on future PRs and our own work.  For example, on this file you quoted, we could return the two-dive component, and the external file does the extra dives, for example?

Reply:
yes i guess we can do that on this file, lets track this on a tech debt issue ?